addons/delivery/models/delivery_carrier.py

```diff
@@ -123,3 +145,129 @@ def cancel_shipment(self, pickings):
         self.ensure_one()
         if hasattr(self, '%s_cancel_shipment' % self.delivery_type):
             return getattr(self, '%s_cancel_shipment' % self.delivery_type)(pickings)
+
+    @api.onchange('state_ids')
+    def onchange_states(self):
+        if self.state_ids:
+            self.country_ids = [(6, 0, self.country_ids.ids + self.state_ids.mapped('country_id.id'))]
+
+    @api.multi
+    def carrier_get(self, contact):
+        self.ensure_one()
+        country_ids = self.country_ids.ids
+        state_ids = self.state_ids.ids
+        if country_ids and not contact.country_id.id in country_ids:
+            return False
+        if state_ids and not contact.state_id.id in state_ids:
+            return False
+        if self.zip_from and (contact.zip or '') < self.zip_from:
+            return False
+        if self.zip_to and (contact.zip or '') > self.zip_to:
+            return False
+        return self
+
+    @api.multi
+    def create_price_rules(self):
+        PriceRule = self.env['delivery.price.rule']
+        for record in self:
+            # if using advanced pricing per destination: do not change
+            if record.delivery_type == 'base_on_rule':
+                continue
+
+            # not using advanced pricing per destination: override lines
+            if record.delivery_type == 'base_on_rule' and not (record.fixed_price is not False or record.free_if_more_than):
+                record.price_rule_ids.unlink()
+
+            # Check that float, else 0.0 is False
+            if not (record.fixed_price is not False or record.free_if_more_than):
+                continue
+
+            if record.delivery_type == 'fixed':
+                PriceRule.search([('carrier_id', '=', record.id)]).unlink()
+
+            #create the delivery price rules
+            if record.free_if_more_than:
+                line_data = {
+                    'carrier_id': record.id,
+                    'name': _('Free if more than %.2f') % record.amount,
```

Comment:
please do not duplicate stuff.

declare line_data before if statement with common values